lib/rules/no-eol-whitespace/index.js

```diff
@@ -34,17 +34,10 @@ function fixString(str) {
 /**
  * @param {number} lastEOLIndex
  * @param {string} string
- * @param {{ ignoreEmptyLines?: boolean, isRootFirst?: boolean }} [options]
+ * @param {{ ignoreEmptyLines: boolean, isRootFirst: boolean }} options
  * @returns {number}
  */
-function findErrorStartIndex(
-	lastEOLIndex,
-	string,
-	{ ignoreEmptyLines, isRootFirst } = {
-		ignoreEmptyLines: false,
-		isRootFirst: false,
-	},
-) {
+function findErrorStartIndex(lastEOLIndex, string, { ignoreEmptyLines, isRootFirst }) {
```

Comment:
[note] `findErrorStartIndex` is a private function in the module, so I've made the argument type simpler instead of fixing a type annotation.

Reply:
I really appreciate the clarity of the notes in your pull requests, @ybiquitous. They're always fantastic.

Reply:
I always appreciate your thoughtful and generous reviews. 😊